src/index.rs

```diff
@@ -231,8 +235,70 @@ impl IndexTable {
 		Ok(try_io!(Ok(&map[offset..offset + CHUNK_LEN])))
 	}
 
-	#[inline(never)]
 	fn find_entry(&self, key_prefix: u64, sub_index: usize, chunk: &[u8]) -> (Entry, usize) {
+		#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+		if is_x86_feature_detected!("sse2") {
```

Comment:
I'd use compile time rather than runtime detection. Similar to how's it [done in `HashMap` ](https://github.com/rust-lang/hashbrown/blob/master/src/raw/mod.rs#L22). I mean we use `HashMap` a lot in this code, and if it breaks at runtime because `target_arch` was incorrect, we're screwed anyway.

Reply:
Done. Indeed.